cmd/soroban-rpc/lib/preflight/src/ledger_storage.rs

```diff
@@ -227,28 +245,27 @@ impl LedgerStorage {
 
 impl SnapshotSource for LedgerStorage {
     fn get(&self, key: &Rc<LedgerKey>) -> Result<(Rc<LedgerEntry>, Option<u32>), HostError> {
-        let mut entry_and_expiration =
-            <LedgerStorage>::get(self, key, self.restore_tracker.is_some())?;
         if let Some(ref tracker) = self.restore_tracker {
+            let mut entry_and_expiration = self
+                .get_including_persistent_expired(key)
+                .map_err(HostError::from)?;
             // If the entry expired, we modify the expiration to make it seem like it was restored
             entry_and_expiration.1 =
                 tracker.track_and_restore(self.current_ledger_sequence, key, &entry_and_expiration);
+            return Ok((entry_and_expiration.0.into(), entry_and_expiration.1));
         }
-        Ok((entry_and_expiration.0.into(), entry_and_expiration.1))
+        let entry_and_expiration =
```

Comment:
Just wanted to double-check: what is the error that occurs when 'auto-restoration' is not enabled and the persistent entry has expired? We need to make sure it's not `NotFound` as to distinguish from a simply non-existent entry.

Reply:
The error is also`NotFound`. (`Missing` for the host).

Yeah, we should probably use a different error code but:
1. This is for a separate PR.
2. I don't think the HostError has a suitable error to express it

Reply:
Actually, I bit the bullet and included a separate error code for expired entries

Reply:
Thanks for addressing this as well.